agent/src/softwarecontaineragent.cpp

```diff
@@ -91,6 +94,47 @@ SoftwareContainerAgent::~SoftwareContainerAgent()
 {
 }
 
+/* we are fetching lxcpath from config now. But in near future we should use lxcConfigPath instead
+ * */
+void SoftwareContainerAgent::removeOldContainers()
+{
+    char **containerNames = nullptr;
+    struct lxc_container **containerList = nullptr;
+
+    const char *basePath = lxc_get_global_config_item("lxc.lxcpath");
+    auto num = list_all_containers(basePath, &containerNames, &containerList);
+    if (num > 0 && num < std::numeric_limits<int>::max()) {
```

Comment:
Wouldn't you rather want to change this to check for -1 (the error case) and otherwise just use the for loop to iterate over the found containers?